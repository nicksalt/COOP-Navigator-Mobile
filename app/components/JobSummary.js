import React from "react";
import {
  View,
  ScrollView,
  Text,
  TouchableOpacity,
  ActivityIndicator
} from "react-native";

export default class JobSummary extends React.Component {
  applyToJob = () => {
    this.props.createJobApplicationDispatch(this.props.jobId);
    this.forceUpdate();
  };

  render() {
    if (this.props.isFetching) {
      return (
        <View>
          <ActivityIndicator size="large" color="gray" />
        </View>
      );
    }

    if (this.props.errorOccurred) {
      return (
        <View>
          <Text>There was an error fetching the job:</Text>
          <Text>{this.props.error.message}</Text>
        </View>
      );
    }

    return (
      <ScrollView>
        <Text style={styles.subtitle}>Job Summary</Text>
        <View style={styles.container}>
          <Text style={styles.field}>Job Number:</Text>
          <Text style={styles.fieldValue}>{this.props.job.jobId}</Text>
          <Text style={styles.field}>Job Term:</Text>
          <Text style={styles.fieldValue}>{this.props.job.term}</Text>
          <Text style={styles.field}>Job Round:</Text>
          <Text style={styles.fieldValue}>{this.props.job.openForRound}</Text>
          <Text style={styles.field}>Number of positions:</Text>
          <Text style={styles.fieldValue}>
            {this.props.job.numberOfPositions}
          </Text>
          <Text style={styles.field}>Title</Text>
          <Text style={styles.fieldValue}>{this.props.job.englishTitle}</Text>
          <Text style={styles.field}>Organization</Text>
          <Text style={styles.fieldValue}>{this.props.job.organization}</Text>
        </View>
        <Text style={styles.subtitle}>Details</Text>
        <View style={styles.container}>
          <Text style={styles.field}>Job status:</Text>
          <Text style={styles.fieldValue}>{this.props.job.status}</Text>
          <Text style={styles.field}>Duration (months):</Text>
          <Text style={styles.fieldValue}>{this.props.job.duration}</Text>
          <Text style={styles.field}>Salary:</Text>
          <Text style={styles.fieldValue}>{this.props.job.salary}</Text>
        </View>
        <Text style={styles.subtitle}>Description</Text>
        <View style={styles.container}>
          <Text style={styles.fieldValue}>
            {this.props.job.englishDescription}
          </Text>
        </View>
        <Text style={styles.subtitle}>Job Qualification</Text>
        <View style={styles.container}>
          <Text style={styles.field}>Job Language:</Text>
          <Text style={styles.fieldValue}>{this.props.job.language}</Text>
          <Text style={styles.field}>Security Clearence:</Text>
          <Text style={styles.fieldValue}>
            {this.props.job.securityClearance}
          </Text>
          <Text style={styles.field}>Minimum CGPA:</Text>
          <Text style={styles.fieldValue}>{this.props.job.minimumCPGA}</Text>
          <Text style={styles.field}>Preferred Program of Studies:</Text>
          <Text style={styles.fieldValue}>
            {this.props.job.preferredProgramsOfStudy.join(", ")}
          </Text>
          <TouchableOpacity
            style={styles.buttonContainer}
            onPress={this.applyToJob}
            title="Apply"
          >
            <Text style={styles.applyButton}>APPLY</Text>
          </TouchableOpacity>

          {this.props.applicationSuccessful && (
            <Text style={styles.centerMsg}>Application successful</Text>
          )}
          {this.props.applicationFailed && (
            <Text style={styles.centerMsg}>{this.props.error.message}</Text>
          )}
        </View>
      </ScrollView>
    );
  }
}

const styles = {
  subtitle: {
    padding: 10,
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
    backgroundColor: "#a9343a"
  },
  container: {
    padding: 10
  },
  field: {
    fontSize: 16,
    fontWeight: "bold"
  },
  fieldValue: {
    fontSize: 15
  },
  buttonContainer: {
    margin: 50,
    borderRadius: 15,
    backgroundColor: "#d8d8d8",
    padding: 15
  },
  applyButton: {
    textAlign: "center",
    fontWeight: "bold",
    color: "#a9343a"
  },
  centerMsg: {
    padding: 5,
    fontSize: 13,
    fontWeight: "bold",
    textAlign: "center",
    color: "black",
    margin: 2
  }
};
