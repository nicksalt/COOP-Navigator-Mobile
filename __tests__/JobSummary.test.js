import React from "react";
import JobSummary from "../app/components/JobSummary";
import renderer from "react-test-renderer";

jest.mock("ScrollView", () => "ScrollView");

test("JobSummary snapshot renders correctly", () => {
  let props = {};
  props.fetchJobSummaryDispatch = () => {};
  props.isFetching = true;
  let tree = renderer.create(<JobSummary {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.isFetching = false;
  props.errorOccurred = true;
  props.error = { message: "ERROR OCCURRED" };
  tree = renderer.create(<JobSummary {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.errorOccurred = false;
  props.applicationSuccessful = true;
  props.job = {
    jobId: 100,
    term: "2018, Summer",
    openForRound: "1",
    numberOfPositions: 4,
    employerJobNumber: "22A",
    englishTitle: "Software Developer/Engineer",
    frenchTitle: "DÃ©veloppeur de logiciels",
    organization: "Apple",
    organizationWebsite: "",
    division: "",
    expiryDate: "Fri Apr 13 2018 23:59:00 GMT-0400 (Eastern Daylight Time)",
    status: "active",
    duration: 4,
    salary: "35$/hour",
    frenchDescription:
      "100 FRENCH Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum",
    englishDescription:
      "100  ENGLISH Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temporincididunt ut labore et dolore magna aliqua. Ut enimad minim veniam, quis nostrud exercitation ullamco laborisnisi ut aliquip ex ea commodo consequat. Duis aute iruredolor in reprehenderit in voluptate velit esse cillum doloreeu fugiat nulla pariatur. Excepteur sint occaecat cupidatatnon proident, sunt in culpa qui officia deserunt mollit animid est laborum",
    language: "English or Bilingual",
    securityClearance: "Level 2 (secret)",
    minimumCPGA: "9",
    requirements: "",
    address: {
      country: "Canada",
      stateOrProvince: "Ontario",
      city: "Ottawa",
      street: "124 Bank Street",
      postalCode: "A1B2C3"
    },
    preferredProgramsOfStudy: ["computer engineering", "software engineering"]
  };
  tree = renderer.create(<JobSummary {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
  props.applicationSuccessful = false;
  props.applicationFailed = true;
  tree = renderer.create(<JobSummary {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
