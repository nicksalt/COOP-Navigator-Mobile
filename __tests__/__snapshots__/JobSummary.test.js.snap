// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobSummary snapshot renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="gray"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`JobSummary snapshot renders correctly 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    There was an error fetching the job:
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ERROR OCCURRED
  </Text>
</View>
`;

exports[`JobSummary snapshot renders correctly 3`] = `
<ScrollView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Job Summary
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Number:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      100
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Term:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      2018, Summer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Round:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Number of positions:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Software Developer/Engineer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Organization
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Apple
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Details
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job status:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      active
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Duration (months):
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Salary:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      35$/hour
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Description
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      100  ENGLISH Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temporincididunt ut labore et dolore magna aliqua. Ut enimad minim veniam, quis nostrud exercitation ullamco laborisnisi ut aliquip ex ea commodo consequat. Duis aute iruredolor in reprehenderit in voluptate velit esse cillum doloreeu fugiat nulla pariatur. Excepteur sint occaecat cupidatatnon proident, sunt in culpa qui officia deserunt mollit animid est laborum
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Job Qualification
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Language:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      English or Bilingual
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Security Clearence:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Level 2 (secret)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Minimum CGPA:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      9
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Preferred Program of Studies:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      computer engineering, software engineering
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#d8d8d8",
          "borderRadius": 15,
          "margin": 50,
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#a9343a",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        APPLY
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
          "margin": 2,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      Application successful
    </Text>
  </View>
</ScrollView>
`;

exports[`JobSummary snapshot renders correctly 4`] = `
<ScrollView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Job Summary
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Number:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      100
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Term:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      2018, Summer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Round:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Number of positions:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Software Developer/Engineer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Organization
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Apple
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Details
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job status:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      active
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Duration (months):
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Salary:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      35$/hour
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Description
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      100  ENGLISH Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temporincididunt ut labore et dolore magna aliqua. Ut enimad minim veniam, quis nostrud exercitation ullamco laborisnisi ut aliquip ex ea commodo consequat. Duis aute iruredolor in reprehenderit in voluptate velit esse cillum doloreeu fugiat nulla pariatur. Excepteur sint occaecat cupidatatnon proident, sunt in culpa qui officia deserunt mollit animid est laborum
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Job Qualification
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Job Language:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      English or Bilingual
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Security Clearence:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Level 2 (secret)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Minimum CGPA:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      9
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Preferred Program of Studies:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      computer engineering, software engineering
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#d8d8d8",
          "borderRadius": 15,
          "margin": 50,
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#a9343a",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        APPLY
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
          "margin": 2,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      ERROR OCCURRED
    </Text>
  </View>
</ScrollView>
`;
